File: tools/patch_arcade_ui.py
path = r'C:/Users/appbu/StudioProjects/ngmy/lib/main.dart'
with open(path, 'r', encoding='utf-8') as f:
    lines = f.readlines()
start = next(i for i, l in enumerate(lines) if "if (widget.gameId == 'bubble_pop')" in l)
end = next(i for i, l in enumerate(lines) if "if (widget.gameId == 'lucky_doors')" in l)
close = end
depth = 0
for j in range(end, min(end + 40, len(lines))):
    depth += lines[j].count('{') - lines[j].count('}')
    if j > end and lines[j].strip() == '}' and depth <= 0:
        close = j
        break
insert = """    if (_isArcade && _arcade != null) {
      return buildNgmyArcadeBoard(
        gameId: widget.gameId,
        state: _arcade!,
        onChanged: () => setState(() {}),
        onFullWin: () => unawaited(_payoutWin(subtitle: 'Arcade goal reached!')),
      );
    }

"""
new_lines = lines[:start] + [insert] + lines[close + 1 :]
with open(path, 'w', encoding='utf-8', newline='\n') as f:
    f.writelines(new_lines)
print(f'removed {close - start + 1} lines; inserted arcade block at line {start + 1}')

File: tools/patch_arcade_ui.dart
import 'dart:io';

void main() {
  final path = 'lib/main.dart';
  final lines = File(path).readAsLinesSync();
  final start = lines.indexWhere((l) => l.contains("__removed_tap_order"));
  final end = lines.indexWhere((l) => l.contains("if (widget.gameId == 'lucky_doors')"));
  if (start < 0 || end < 0) {
    stderr.writeln('markers not found start=$start end=$end');
    exit(1);
  }
  var close = end;
  var depth = 0;
  for (var j = end; j < lines.length && j < end + 40; j++) {
    depth += '{'.allMatches(lines[j]).length - '}'.allMatches(lines[j]).length;
    if (j > end && lines[j].trim() == '}' && depth <= 0) {
      close = j;
      break;
    }
  }
  final out = [...lines.sublist(0, start), ...lines.sublist(close + 1)];
  File(path).writeAsStringSync('${out.join('\n')}\n');
  stdout.writeln('Removed lines ${start + 1}-${close + 1}');
}
